doc/how_to/custom_components/esm/panes.md

```diff
@@ -0,0 +1,178 @@
+# Create Panes with ReactiveHTML
+
+In this guide we will show you how to efficiently implement custom panes using `JSComponent`, `ReactComponent` and `AnyWidgetComponent` to get input from the user.
+
+## Creating a ChartJS Pane
+
+This example will show you the basics of creating a [ChartJS](https://www.chartjs.org/docs/latest/) pane.
+
+```{pyodide}
+import panel as pn
+import param
+from panel.custom import JSComponent
+
+
+class ChartJSComponent(JSComponent):
+    object = param.Dict()
+
+    _esm = """
+import { Chart } from "https://esm.sh/chart.js/auto"
+
+let chart = null;
+
+function createChart(canvasEl, model) {
+    if (chart) {
+        chart.destroy();
+    }
+    chart = new Chart(canvasEl.getContext('2d'), model.object);
+}
+
+export function render({ model }) {
+    const canvasEl = document.createElement('canvas');
+    // The chart will not render before after the layout has been created
+    requestAnimationFrame(() => createChart(canvasEl, model));
```

Comment:
The `ReactiveHTML`  creates the chart in `after_layout`. There is nothing similar here. Postponing the creating via requestAnimationFrame seems to do the trick. Without it Chart.js won't render.

What is the correct way to do this?

Reply:
`model.on('after_render', ...)` or more explicitly `model.on('lifecycle:after_render', ...)`

Reply:
works in the example